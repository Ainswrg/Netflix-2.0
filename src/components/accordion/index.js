/* eslint-disable no-shadow */
import React, { createContext, useContext, useState } from 'react';
import { Container, Frame, Title, Item, Inner, Header, Body } from './styles/accordion';

const ToggleContext = createContext();

export default function Accordion({ children, ...props }) {
  return (
    <Container {...props}>
      <Inner>{children}</Inner>
    </Container>
  );
}

Accordion.Title = function AccordionTitle({ children, ...props }) {
  return <Title {...props}>{children}</Title>;
};

Accordion.Frame = function AccordionFrame({ children, ...props }) {
  return <Frame {...props}>{children}</Frame>;
};

Accordion.Item = function AccordionItem({ children, ...props }) {
  const [toggleShow, setToggleShow] = useState(false);

  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...props}>{children}</Item>
    </ToggleContext.Provider>
  );
};

Accordion.Header = function AccordionHeader({ children, ...props }) {
  const { toggleShow, setToggleShow } = useContext(ToggleContext);

  return (
    <Header clicked={toggleShow} onClick={() => setToggleShow(!toggleShow)} {...props}>
      {children}
    </Header>
  );
};

Accordion.Body = function AccordionBody({ children, ...props }) {
  const { toggleShow } = useContext(ToggleContext);

  return toggleShow ? <Body {...props}>{children}</Body> : null;
};
