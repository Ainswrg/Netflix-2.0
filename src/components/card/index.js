import { createContext, useContext, useState } from 'react';

import {
  Entities,
  Container,
  Group,
  Item,
  Title,
  SubTitle,
  Text,
  Feature,
  FeatureTitle,
  FeatureText,
  FeatureClose,
  Maturity,
  Content,
  Meta,
  Image,
} from './styles/card';

export const FeatureContext = createContext();

export default function Card({ children, ...props }) {
  const [showFeature, setShowFeature] = useState(false);
  const [itemFeature, setItemFeature] = useState({});

  return (
    <FeatureContext.Provider value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}>
      <Container {...props}>{children}</Container>
    </FeatureContext.Provider>
  );
}

Card.Group = function CardGroup({ children, ...props }) {
  return <Group {...props}>{children}</Group>;
};

Card.Title = function CardTitle({ children, ...props }) {
  return <Title {...props}>{children}</Title>;
};

Card.SubTitle = function CardSubTitle({ children, ...props }) {
  return <SubTitle {...props}>{children}</SubTitle>;
};

Card.Text = function CardText({ children, ...props }) {
  return <Text {...props}>{children}</Text>;
};

Card.Entities = function CardEntities({ children, ...props }) {
  return <Entities {...props}>{children}</Entities>;
};

Card.Meta = function CardMeta({ children, ...props }) {
  return <Meta {...props}>{children}</Meta>;
};

Card.Item = function CardItem({ item, children, ...props }) {
  // const { setShowFeature, setItemFeature } = useContext(FeatureContext);

  return (
    <Item
      // onClick={() => {
      //   setItemFeature(item);
      //   setShowFeature(true);
      // }}
      {...props}
    >
      {children}
    </Item>
  );
};

Card.Image = function CardImage({ ...props }) {
  return <Image {...props} />;
};
