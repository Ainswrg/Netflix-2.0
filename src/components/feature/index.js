import { Container, Title, SubTitle } from './styles/feature';

export default function Feature({ children, ...props }) {
  return <Container {...props}>{children}</Container>;
}

Feature.Title = function FeatureTitle({ children, ...props }) {
  return <Title {...props}>{children}</Title>;
};

Feature.SubTitle = function FeatureSubTitle({ children, ...props }) {
  return <SubTitle {...props}>{children}</SubTitle>;
};
