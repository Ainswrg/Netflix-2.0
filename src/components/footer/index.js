import React from 'react';
import { Container, Row, Column, Link, Title, Text, Break } from './styles/footer';

export default function Footer({ children, ...props }) {
  return <Container {...props}> {children} </Container>;
}

Footer.Row = function FooterRow({ children, ...props }) {
  return <Row {...props}> {children} </Row>;
};

Footer.Column = function FooterColumn({ children, ...props }) {
  return <Column {...props}> {children} </Column>;
};

Footer.Link = function FooterLink({ children, ...props }) {
  return <Link {...props}> {children} </Link>;
};

Footer.Title = function FooterTitle({ children, ...props }) {
  return <Title {...props}> {children} </Title>;
};

Footer.Text = function FooterText({ children, ...props }) {
  return <Text {...props}> {children} </Text>;
};

Footer.Break = function FooterBreak({ children, ...props }) {
  return <Break {...props}> {children} </Break>;
};
