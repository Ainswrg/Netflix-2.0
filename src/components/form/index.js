import { Container, Error, Title, Text, TextSmall, Link, Input, Submit, Base } from './styles/form';

export default function Form({ children, ...props }) {
  return <Container {...props}>{children}</Container>;
}

Form.Base = function FormBase({ children, ...props }) {
  return <Base {...props}>{children}</Base>;
};

Form.Error = function FormError({ children, ...props }) {
  return <Error {...props}>{children}</Error>;
};

Form.Text = function FormText({ children, ...props }) {
  return <Text {...props}>{children}</Text>;
};

Form.TextSmall = function FormTextSmall({ children, ...props }) {
  return <TextSmall {...props}>{children}</TextSmall>;
};

Form.Title = function FormTitle({ children, ...props }) {
  return <Title {...props}>{children}</Title>;
};

Form.Link = function FormLink({ children, ...props }) {
  return <Link {...props}>{children}</Link>;
};

Form.Input = function FormInput({ children, ...props }) {
  return <Input {...props}>{children}</Input>;
};

Form.Submit = function FormSubmit({ children, ...props }) {
  return <Submit {...props}>{children}</Submit>;
};
