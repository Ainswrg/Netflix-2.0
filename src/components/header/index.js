import { useState } from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';

import {
  Background,
  ButtonLink,
  Container,
  Dropdown,
  FadeBottom,
  Feature,
  FeatureCallOut,
  Gradient,
  Group,
  Link,
  Logo,
  Navbar,
  Picture,
  PlayButton,
  Profile,
  Search,
  SearchIcon,
  SearchInput,
  Text,
} from './styles/header';

export default function Header({ bg = true, children, ...props }) {
  return bg ? <Background {...props} data-testid="header-bg">{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...props }) {
  return <Container {...props}>{children}</Container>;
};

Header.Group = function HeaderGroup({ children, ...props }) {
  return <Group {...props}>{children}</Group>;
};

Header.SearchTerm = function HeaderSearch({ searchTerm, setSearchTerm, setCategory, ...props }) {
  const [searchActive, setSearchActive] = useState(false);

  return (
    <Search {...props}>
      <SearchIcon
        onClick={() => {
          setSearchActive(!searchActive);
        }}
        data-testid="search-click"
      >
        <img src="/images/icons/search.png" alt="search" />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder="Search films and series"
        active={searchActive}
        data-testid="search-input"
      />
    </Search>
  );
};

Header.PlayButton = function HeaderPlayButton({ children, ...props }) {
  return <PlayButton {...props}>{children}</PlayButton>;
};

Header.Dropdown = function HeaderDropdown({ children, ...props }) {
  return <Dropdown {...props}>{children}</Dropdown>;
};

Header.Feature = function HeaderFeature({ children, ...props }) {
  return <Feature {...props}>{children}</Feature>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...props }) {
  return <FeatureCallOut {...props}>{children}</FeatureCallOut>;
};

Header.Profile = function HeaderProfile({ children, ...props }) {
  return <Profile {...props}>{children}</Profile>;
};

Header.Text = function HeaderText({ children, ...props }) {
  return <Text {...props}>{children}</Text>;
};

Header.Picture = function HeaderPicture({ src, ...props }) {
  return <Picture {...props} src={`images/users/${src}.png`} />;
};

Header.TextLink = function HeaderTextLink({ children, ...props }) {
  return <Link {...props}>{children}</Link>;
};

Header.Logo = function HeaderLogo({ to, ...props }) {
  return (
    <ReactRouterLink to={to} {...props}>
      <Logo {...props} />
    </ReactRouterLink>
  );
};

Header.ButtonLink = function HeaderButtonLink({ children, ...props }) {
  return <ButtonLink {...props}>{children}</ButtonLink>;
};

Header.Navbar = function HeaderNavbar({ children, ...props }) {
  return <Navbar {...props}> {children} </Navbar>;
};

Header.Gradient = function HeaderGradient({ ...props }) {
  return <Gradient {...props} />;
};

Header.FadeBottom = function HeaderFadeBottom({ ...props }) {
  return <FadeBottom {...props} />;
};
