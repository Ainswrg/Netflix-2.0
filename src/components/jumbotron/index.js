import {
  Container,
  Inner,
  Item,
  Title,
  SubTitle,
  Pane,
  Image,
  Video,
} from "./styles/jumbotron";

export const Jumbotron = ({ children, direction = "row", ...props }) => {
  return (
    <Item {...props}>
      <Inner direction={direction}>{children}</Inner>
    </Item>
  );
};

Jumbotron.Container = function JumbotronContainer({ children, ...props }) {
  return <Container {...props}>{children}</Container>;
};

Jumbotron.Pane = function JumbotronPane({ children, ...props }) {
  return <Pane {...props}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle({ children, ...props }) {
  return <Title {...props}>{children}</Title>;
};

Jumbotron.SubTitle = function JumbotronSubTitle({ children, ...props }) {
  return <SubTitle {...props}>{children}</SubTitle>;
};

Jumbotron.Image = function JumbotronImage({ children, ...props }) {
  return <Image {...props} />;
};
Jumbotron.Video = function JumbotronVideo({ children, ...props }) {
  return <Video autoPlay loop muted playsInline {...props} />;
};
