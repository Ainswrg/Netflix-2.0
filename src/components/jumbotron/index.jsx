import React from 'react';
import { Container, Inner, Item, Title, SubTitle, Pane, Image, Video, Flex } from './styles/jumbotron';

export default function Jumbotron({ children, direction = 'row', ...props }) {
  return (
    <Item {...props}>
      <Inner direction={direction}> {children} </Inner>
    </Item>
  );
}

Jumbotron.Container = function JumbotronContainer({ children, ...props }) {
  return <Container {...props}>{children}</Container>;
};

Jumbotron.Pane = function JumbotronPane({ children, ...props }) {
  return <Pane {...props}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle({ children, ...props }) {
  return <Title {...props}> {children} </Title>;
};

Jumbotron.SubTitle = function JumbotronSubTitle({ children, ...props }) {
  return <SubTitle {...props}> {children} </SubTitle>;
};

Jumbotron.Image = function JumbotronImage({ ...props }) {
  return <Image {...props} />;
};

Jumbotron.Video = function JumbotronVideo({ ...props }) {
  return <Video autoPlay loop muted playsInline {...props} />;
};

Jumbotron.Flex = function JumbotronFlex({ children, ...props }) {
  return <Flex {...props}> { children } </Flex>;
};
