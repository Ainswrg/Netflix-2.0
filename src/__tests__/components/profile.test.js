/* eslint-disable no-undef */
import { render } from '@testing-library/react';
import { Profiles } from '../../components';

describe('<Profiles />', () => {
  it('renders the <Profiles /> with populated data', () => {
    const { container, getByText, findByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture src="/images.jaslan.png" data-testid="profile-picture" />
            <Profiles.Name>Jaslan D</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );

    expect(getByText("Who's watching?"));
    expect(findByTestId('profile-picture')).toBeTruthy();
    expect(getByText('Jaslan D')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Profiles /> with populated data but misc profile image', () => {
    const { container, getByText, findByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture data-testid="profile-picture-mics" />
            <Profiles.Name>Jaslan D</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );

    expect(getByText("Who's watching?"));
    expect(findByTestId('profile-picture-misc')).toBeTruthy();
    expect(getByText('Jaslan D')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
