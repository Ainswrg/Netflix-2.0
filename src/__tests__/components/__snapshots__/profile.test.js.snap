// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders the <Profiles /> with populated data 1`] = `
<div
  class="profiles__Container-sc-bxtkkv-0 kVDfqY"
>
  <h1
    class="profiles__Title-sc-bxtkkv-1 jNMSoS"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-sc-bxtkkv-2 cMxYXf"
  >
    <li
      class="profiles__Item-sc-bxtkkv-5 hGFeax"
    >
      <img
        class="profiles__Picture-sc-bxtkkv-4 kvvZaQ"
        src="/images/users//images.jaslan.png.png"
      />
      <p
        class="profiles__Name-sc-bxtkkv-3 gkICWz"
      >
        Jaslan D
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profiles /> renders the <Profiles /> with populated data but misc profile image 1`] = `
<div
  class="profiles__Container-sc-bxtkkv-0 kVDfqY"
>
  <h1
    class="profiles__Title-sc-bxtkkv-1 jNMSoS"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-sc-bxtkkv-2 cMxYXf"
  >
    <li
      class="profiles__Item-sc-bxtkkv-5 hGFeax"
    >
      <img
        class="profiles__Picture-sc-bxtkkv-4 kvvZaQ"
        src="/images/misc/loading.gif"
      />
      <p
        class="profiles__Name-sc-bxtkkv-3 gkICWz"
      >
        Jaslan D
      </p>
    </li>
  </ul>
</div>
`;
